### PR TITLE
#163143276 Paginate the wings response

### PR DESCRIPTION
**Pull request by kampkelly**
#### What does this PR do?
- Paginate the wings response

#### Description of Task to be completed?
The task is to allow the admin to be able to scroll through all the available wings .

#### How should this be manually tested?
- Run the server `http://127.0.0.1:5000/mrm`.
- Run the following query
```
query {
    allWings(page: 1, perPage: 2) {
        wings {
            id
            name
            floorId
        }
        queryTotal
        pages
        hasNext
        hasPrevious
    }
}
```

**Any background context you want to provide?**
None

**What are the relevant pivotal tracker stories?**
[#163143276](https://www.pivotaltracker.com/story/show/163143276)

#### Screenshots
screen shot 2019-01-14 at 11 53 08 am

**Questions:**
None

### CODE REVIEW

File: api/wing/schema.py [hunk 1]
```diff
@@ -73,12 +74,37 @@ def mutate(self, info, wing_id, **kwargs):
...
         query = Wing.get_query(info)
-        return query.all()
+        if not page:
+            return query.all()
+        page = validate_page(page)
+        self.query_total = query.count()
+        result = query.limit(per_page).offset(page*per_page)
+        if result.count() == 0:
+            return GraphQLError("No more resources")
+        return result
```
Comment (caveinn) [inline api/wing/schema.py]: Isn't it better to raise an error rather than return an error?

Comment (bevkololi) [reply api/wing/schema.py]: +1

File: api/wing/schema.py [hunk 2]
```diff
@@ -73,12 +74,37 @@ def mutate(self, info, wing_id, **kwargs):
...
+        per_page = self.per_page
         query = Wing.get_query(info)
-        return query.all()
+        if not page:
+            return query.all()
+        page = validate_page(page)
+        self.query_total = query.count()
+        result = query.limit(per_page).offset(page*per_page)
+        if result.count() == 0:
+            return GraphQLError("No more resources")
```
Comment (cwizard2011) [inline api/wing/schema.py]: Please implement the feedback given by @bevkololi and @caveinn , `raise GraphQLError`

Comment (cwizard2011) [inline api/wing/schema.py]: Can you use a more descriptive message here? something like `No wing on this page` or something better, considering we also have resources schema in the application, this error message can be confusing

File: api/wing/schema.py (hunk 1 above)
Comment (kampkelly) [reply api/wing/schema.py]: This has been changed now. Thanks.

File: api/wing/schema.py [hunk 3]
```diff
@@ -73,12 +74,37 @@ def mutate(self, info, wing_id, **kwargs):
...
+        per_page = self.per_page
         query = Wing.get_query(info)
-        return query.all()
+        if not page:
+            return query.all()
+        page = validate_page(page)
+        self.query_total = query.count()
+        result = query.limit(per_page).offset(page*per_page)
+        if result.count() == 0:
+            raise GraphQLError("No more resources")
```
Comment (cwizard2011) [inline api/wing/schema.py]: Can you use a more descriptive message here? something like No wing on this page or something better, considering we also have resources schema in the application, this error message can be confusing

Comment (kampkelly) [reply api/wing/schema.py]: @cwizard2011 I agree with you. This has now been changed to `No more wings` as it is more descriptive.

File: api/wing/schema.py (hunk 2 above)
Comment (kampkelly) [reply api/wing/schema.py]: The feedback has been implemented.

**Comment by joshuaocero** [comment]
This feature is no longer necessary as we shall be restructuring the room structure to have wing information as tags

**Pull request closed without merging**